internal/gengapic/gengapic.go

```diff
@@ -439,10 +442,30 @@ func (g *generator) emptyUnaryCall(servName string, m *descriptor.MethodDescript
 	return nil
 }
 
-func (g *generator) insertMetadata() {
+func (g *generator) insertMetadata(m *descriptor.MethodDescriptorProto) {
+	if headers := parseRequestHeaders(m); len(headers) > 0 {
+		field := headers[0][1]
+		pairs := pair("x-goog-request-params", field)
+
+		for _, h := range headers[1:] {
+			field = h[1]
+			pairs += fmt.Sprintf(", %s", pair("x-goog-request-params", field))
+		}
+
+		g.printf("md := metadata.Pairs(%s)", pairs)
+		g.printf("ctx = insertMetadata(ctx, c.xGoogMetadata, md)")
+
+		return
+	}
+
 	g.printf("ctx = insertMetadata(ctx, c.xGoogMetadata)")
 }
 
+func pair(key, val string) string {
+	// TODO(ndietz) the formatted field accessor here only works for top-level fields
+	return fmt.Sprintf("%q, fmt.Sprintf(%q, %q, req.Get%s())", key, "%s=%v", val, snakeToCamel(val))
```

Comment:
Nice :P

Reply:
🙃
I haven’t seen an http transcoding option contain a nested field request param...figure i’d deal with it if it ever happens